### PR TITLE
Sync upstream cluster-autoscaler-release-1.26 for patch

### LINKED ISSUE #211: Issue for - Sync with Upstream v1.26.2
## What would you like to be added
Gardener autoscaler should be synced with respective [Kubernetes Autoscaler 1.26.2 Release](https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.26.1)

## Why is this needed

* To keep the fork in sync with upstream patches.

## Steps

Please follow the steps in
[How do I sync gardener autoscaler with an upstream autoscaler patch release](https://github.com/gardener/autoscaler/blob/machine-controller-manager-provider/cluster-autoscaler/FAQ.md#how-do-i-sync-gardener-autoscaler-with-an-upstream-autoscaler-patch-release)

### PR DESCRIPTION
**Pull request by elankath**
**What this PR does / why we need it**:
Syncs the autoscaler with upstream v1.26.2

**Which issue(s) this PR fixes**:
Fixes #211 

**Special notes for your reviewer**:
NOTE: This fixes upstream autoscaler issue: https://github.com/kubernetes/autoscaler/issues/5372

**Release note**:

```other operator
Synced the changes till v1.26.2 of upstream autoscaler release branch: cluster-autoscaler-release-1.26
Includes fix for https://github.com/kubernetes/autoscaler/issues/5372
```

### CODE REVIEW

**Comment by ghost** [comment]
Thank you @elankath for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.26.md [hunk 1]
```diff
@@ -39,12 +39,14 @@
...
 
 ## Changes made
+ - See general release notes of 1.26.2: https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.26.2
  - See general release notes of 1.26.1: https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.26.1
 
 ### During merging
+  - `k8s.io/*` package versions upgraded to newer ones specific in upstream release branch.
 
 ### During vendoring k8s
 - mcm v0.48.1 -> 0.48.2
```
Comment (himanshu-kun) [inline cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.26.md:60]: we can also update to which k8s dependency (k8s 1.26.3) we have upgraded in go.mod

File: builder/Dockerfile [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.19
+FROM golang:1.19.7
```
Comment (himanshu-kun) [inline builder/Dockerfile:15]: kindly also update the golang version in the dockerfile we use to build autoscaler image
https://github.com/elankath/autoscaler/blob/sync-rel-v1.26/cluster-autoscaler/Dockerfile

**Review by himanshu-kun — CHANGES_REQUESTED** [review]
small changes requested.

File: cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.26.md (hunk 1 above)
Comment (elankath) [reply cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.26.md:60]: IT now pass for 1.26.3 . Thanks to Rishabh for assistance.

File: builder/Dockerfile (hunk 2 above)
Comment (elankath) [reply builder/Dockerfile:15]: fixed

**Comment by elankath** [comment]
Integration tests now passing after re-vendoring to 1.26.1

```
...
Ran 11 of 11 Specs in 851.680 seconds
SUCCESS! -- 11 Passed | 0 Failed | 0 Pending | 0 Skipped
```

File: cluster-autoscaler/go.mod [hunk 3]
```diff
@@ -198,60 +198,60 @@ replace github.com/digitalocean/godo => github.com/digitalocean/godo v1.27.0
...
+replace k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.26.1
 
-replace k8s.io/kubectl => k8s.io/kubectl v0.26.0
+replace k8s.io/kubectl => k8s.io/kubectl v0.26.1
 
-replace k8s.io/kubelet => k8s.io/kubelet v0.26.0
+replace k8s.io/kubelet => k8s.io/kubelet v0.26.1
 
-replace k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.26.0
+replace k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.26.1
```
Comment (himanshu-kun) [inline cluster-autoscaler/go.mod]: the solution itself depends upon revendor against 1.26.3
Kindly fix the IT accordingly

**Review by himanshu-kun — CHANGES_REQUESTED** [review]

File: cluster-autoscaler/go.mod (hunk 3 above)
Comment (elankath) [reply cluster-autoscaler/go.mod]: Yes, re-vendored against 1.26.3 and fixed IT. `adjustNodeGroups` and `scaleDown` were throwing issues due to autoscaler moving coredns to test nodes. This was resolved and re-tested twice.

File: cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.26.md (hunk 1 above)
Comment (himanshu-kun) [reply cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.26.md:60]: can you address this :)

**Review by himanshu-kun — CHANGES_REQUESTED** [review]
just one tiny change requested, o.w. all looks good

**Comment by elankath** [comment]
> just one tiny change requested, o.w. all looks good

made changes mentioning upgrades in k8s.io packages in sync_changes. 
latest IT run passed.

**Review by himanshu-kun — APPROVED** [review]
/lgtm

**Pull request merged**